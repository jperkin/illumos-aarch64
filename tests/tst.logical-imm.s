/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.4.4 Logical (immediate)
 */
.macro logical_imm op zreg=0
	\op	w0,  w30, #0x00000001
	\op	w0,  w30, #0x7fffffff
	\op	x0,  x30, #0x00000001
	\op	x0,  x30, #0xffffffff
	.if \zreg
	\op	wzr, wzr, #0x00000001
	\op	wzr, wzr, #0x7fffffff
	\op	xzr, xzr, #0x00000001
	\op	xzr, xzr, #0xffffffff
	.else
	\op	wsp, wzr, #0x00000001
	\op	wsp, wzr, #0x7fffffff
	\op	sp,  xzr, #0x00000001
	\op	sp,  xzr, #0xffffffff
	.endif
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.11 AND (immediate)
	 */
	logical_imm and
	/*
	 * C5.6.141 ORR (immediate)
	 * C5.6.124 MOV (bitmask immediate) (alias)
	 */
	logical_imm orr
	/*
	 * C5.6.64 EOR (immediate)
	 */
	logical_imm eor
	/*
	 * C5.6.13 ANDS (immediate)
	 * C5.6.209 TST (immediate) (alias)
	 */
	logical_imm ands 1
.size libdis_test, [.-libdis_test]
