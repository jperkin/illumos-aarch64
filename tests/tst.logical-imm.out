    libdis_test:      c0 03 00 12  and w0, w30, #0x1
    libdis_test+0x4:  c0 7b 00 12  and w0, w30, #0x7fffffff
    libdis_test+0x8:  c0 03 40 92  and x0, x30, #0x1
    libdis_test+0xc:  c0 7f 40 92  and x0, x30, #0xffffffff
    libdis_test+0x10: ff 03 00 12  and wsp, wzr, #0x1
    libdis_test+0x14: ff 7b 00 12  and wsp, wzr, #0x7fffffff
    libdis_test+0x18: ff 03 40 92  and sp, xzr, #0x1
    libdis_test+0x1c: ff 7f 40 92  and sp, xzr, #0xffffffff
    libdis_test+0x20: c0 03 00 32  orr w0, w30, #0x1
    libdis_test+0x24: c0 7b 00 32  orr w0, w30, #0x7fffffff
    libdis_test+0x28: c0 03 40 b2  orr x0, x30, #0x1
    libdis_test+0x2c: c0 7f 40 b2  orr x0, x30, #0xffffffff
    libdis_test+0x30: ff 03 00 32  mov wsp, #0x1
    libdis_test+0x34: ff 7b 00 32  mov wsp, #0x7fffffff
    libdis_test+0x38: ff 03 40 b2  mov sp, #0x1
    libdis_test+0x3c: ff 7f 40 b2  mov sp, #0xffffffff
    libdis_test+0x40: c0 03 00 52  eor w0, w30, #0x1
    libdis_test+0x44: c0 7b 00 52  eor w0, w30, #0x7fffffff
    libdis_test+0x48: c0 03 40 d2  eor x0, x30, #0x1
    libdis_test+0x4c: c0 7f 40 d2  eor x0, x30, #0xffffffff
    libdis_test+0x50: ff 03 00 52  eor wsp, wzr, #0x1
    libdis_test+0x54: ff 7b 00 52  eor wsp, wzr, #0x7fffffff
    libdis_test+0x58: ff 03 40 d2  eor sp, xzr, #0x1
    libdis_test+0x5c: ff 7f 40 d2  eor sp, xzr, #0xffffffff
    libdis_test+0x60: c0 03 00 72  ands w0, w30, #0x1
    libdis_test+0x64: c0 7b 00 72  ands w0, w30, #0x7fffffff
    libdis_test+0x68: c0 03 40 f2  ands x0, x30, #0x1
    libdis_test+0x6c: c0 7f 40 f2  ands x0, x30, #0xffffffff
    libdis_test+0x70: ff 03 00 72  tst wzr, #0x1
    libdis_test+0x74: ff 7b 00 72  tst wzr, #0x7fffffff
    libdis_test+0x78: ff 03 40 f2  tst xzr, #0x1
    libdis_test+0x7c: ff 7f 40 f2  tst xzr, #0xffffffff
